File: hackerrank/insertion-sort/insertion-sort-pt-2/solution.py
def insertionSort2(n, arr):
    for i in range(1, n):
        j = i

        while j > 0 and arr[j] < arr[j - 1]:
            temp = arr[j - 1]
            arr[j - 1] = arr[j]
            arr[j] = temp
            j -= 1

        print(*arr, sep=" ")

File: hackerrank/insertion-sort/insertion-sort-pt-2/solution.ts
function insertionSort2(n: number, arr: number[]): void {
  for (let i = 1; i < n; i++) {
    let j = i;

    while (j > 0 && arr[j] < arr[j - 1]) {
      // swap
      const temp = arr[j - 1];
      arr[j - 1] = arr[j];
      arr[j] = temp;

      j--;
    }

    console.log(arr.join(" "));
  }

  return;
}
